src/hooks/storage/useLocalStorage.ts

```diff
@@ -0,0 +1,43 @@
+import { useState } from 'react';
+
+/**
+ * @description 페이지 새로 고침을 통해 상태가 유지되도록 로컬 저장소에 동기화합니다.
+ *
+ * @param key 로컬 저장소에 저장될 키
+ * @param initialValue 초기 값
+ * @returns [storedValue, setValue] - 로컬 저장소에 저장된 값, 저장 함수
+ */
+function useLocalStorage<T>(key: string, initialValue: T) {
+  // State to store our value
+  // Pass initial state function to useState so logic is only executed once
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    if (typeof window === 'undefined') {
+      return initialValue;
+    }
+    try {
+      const item = window.localStorage.getItem(key);
+
+      return item ? JSON.parse(item) : initialValue;
+    } catch (error) {
+      console.log(error);
+
+      return initialValue;
+    }
```

Comment:
지금 린트상에서 `log`가 워닝이 뜨고, 빌드시 제외되게 해두어서

```suggestion
    } catch (error) {
      console.error(error);

      return initialValue;
    }
```

이렇게 해보면 좋을 거 같아요 ~